### DISCUSSION TITLE
Cpp extension for libyang in master

### DISCUSSION THREAD

**Discussion by srikanths1** [CESNET/libyang#2270] [Q&A] [answered]
Hi, 
For, https://github.com/CESNET/libyang-cpp
The libyang-cpp is dependent on https://github.com/CESNET/libyang/tree/b3d079fc37ac119f5a2a4a120b86d1c7102c3a08 to work, likewise libnetconf2-cpp depends on libyang-cpp, going back to older versions for all packages including sysrepo and netopeer2. 

C++ bindings for libyang
License Gerrit Zuul CI

libyang-cpp implements object-oriented bindings of the [libyang](https://github.com/CESNET/libyang) library in modern C++. Object lifetimes are managed automatically via RAII.

Dependencies
[libyang](https://github.com/CESNET/libyang) - the devel branch (even for the master branch of libyang-cpp)
**temporarily (March 2024) this requires the [pre-v3 API of libyang (commit b3d079fc3)](https://github.com/CESNET/libyang/tree/b3d079fc37ac119f5a2a4a120b86d1c7102c3a08)**
C++20 compiler (e.g., GCC 10.x+, clang 10+)
CMake 3.19+

Is there a plan or any Support to pick this change for main package

**Comment by michalvasko** [comment]
> Is there a plan or any Support to pick this change for main package

What change exactly? I got lost. The C++ bindings are a separate project and we intend to keep it that way.

**Reply by srikanths1** [reply]
libyang cpp version does not build with libyang-3.1.0 and requires the dev branch (3months old ) with above commit, so asking if this commit will be picked up in libyang ?

**Reply by michalvasko** [reply, marked as answer]
What do you mean "picked up", released? The commit exists and can be referenced but should only be used temporarily, then the latest `devel` will again be required. I do not think any action is required in the *libyang* project.

**Comment by svashok** [comment]
Michal, In my understanding Srikanth was trying to build libyang-cpp along with the latest libnetconf/libyang, but it has a temporary dependency on the pre-V3 libyang API. So his question is how to build libyang-cpp with the latest libnetconf/libyang and when will the temporary API dependency be removed?

**Reply by michalvasko** [reply]
To build it, you need to download *libyang* on the specific commit, should not be a problem. To learn when the dependency will be removed, ask in the corresponding project, *libyang-cpp* is a separate and different project than *libyang*, with completely different owners and developers.

**Reply by svashok** [reply]
got it. Thanks Michal
